docs/v3/core/index.rst

```diff
@@ -685,11 +673,16 @@ The following members are optional:
     same dimension name across multiple arrays within the same Zarr hierarchy,
     but extensions or specific applications may do so.
 
-The array metadata object must not contain any other names.
-Those are reserved for future versions of this specification.
-An implementation must fail to open Zarr hierarchies, groups
-or arrays with unknown metadata fields, with the exception of
-objects with a ``"must_understand": false`` key-value pair.
+.. _array-metadata-extensions:
+
+Unknown
+^^^^^^^
+
+All other keys found in the metadata object MUST be interpreted
+following the :ref:`Extensions section <extensions_section>`.
```

Comment:
I think this is misleading here. Array metadata under 3.1 must not have additional keys. Observing additional keys would indicate another version. With our current proposals, extensions cannot add root-level metadata keys.

```suggestion
Other keys are reserved for future versions of the specification.
Therefore, there MUST NOT be other keys in the metadata object.
See :ref:`Extensions section <extensions_section>` for more details.
```

Reply:
Wouldn't that be a regression? Array metadata under 3.0 with an unknown metadata field with ``"must_understand": false`` was valid, and now would be invalid.

Reply:
Fair enough. `"must_understand": false` should still be allowed, but discouraged in favor of explicit extensions.

Reply:
Fair enough. `"must_understand": false` should still be allowed, but discouraged in favor of explicit extensions.

Reply:
Added a short section locally. will push shortly. We may want to expand it.